pkg/tls_extensions/atls_listener.c

```
@@ -0,0 +1,535 @@
+        return NULL;
+    }
+
+    conn->ssl = SSL_new(tls_server->ctx);
+    conn->socket_fd = client_fd;
+    conn->ctx = NULL;
+    SSL_set_fd(conn->ssl, client_fd);
+
+    if (SSL_accept(conn->ssl) <= 0) {
+        perror("unable to accept. SSL handshake failed.\n");
+        free(conn);
+        SSL_free(conn->ssl);
```

cppcheck

[error] Dereferencing 'conn' after it is deallocated / released

(deallocuse)